lib/ansible/modules/cloud/amazon/ssm_parameter_store.py

```diff
@@ -0,0 +1,214 @@
+    description: "This is your first key"
+    value: "World"
+  register: result
+
+- name: Delete the key
+  ssm_parameter_store:
+    name: "Hello"
+    state: absent
+  register: result
+
+- name: Create or update secure key/vaule pair in aws parameter store
+  ssm_parameter_store:
+    name: "Hello"
+    description: "This is your first key"
+    string_type: "SecureString"
+    value: "World"
+  register: result
+
+- name: Retrieving plain-text secret
+  ssm_parameter_store:
+    name: "Hello"
+    state: show
+  register: result
+
+- name: Retrieving plain-text secret with custom kms key
+  ssm_parameter_store:
+    name: "Hello"
+    key_id: "aws/ssm"
+    state: show
+  register: result
+
+- name: Retrieving secret without decrypted
+  ssm_parameter_store:
+    name: "Hello"
+    decryption: False
+    state: show
+  register: result
+'''
+
+try:
+    import botocore
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, AnsibleAWSError, ec2_argument_spec, get_aws_connection_info
+
+def create_update_parameter(client, module):
+    changed = False
+
+    args = dict(
+              Name=module.params.get('name'),
+              Value=module.params.get('value'),
+              Type=module.params.get('string_type'),
+              Overwrite=module.params.get('overwrite')
+    )
+
+    if module.params.get('description'):
+       args.update(Description=module.params.get('description'))
+
+    if module.params.get('string_type') is 'SecureString':
+       args.update(KeyId=module.params.get('key_id'))
+
+    try:
+      nacl = client.put_parameter(**args)
+      changed = True
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+    return changed, nacl
+
+def get_parameter(client, module):
+    changed = False
+    try:
+      nacl = client.get_parameters(
+                Names=[module.params.get('name')],
+                WithDecryption=module.params.get('decryption')
+      )
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+    return changed, nacl['Parameters']
+
+def delete_parameter(client, module):
+    changed = False
+
+    nacl = dict()
+
+    get_nacl = client.get_parameters(
+                  Names=[module.params.get('name')]
+    )
+    if get_nacl['Parameters']:
+      try:
+        nacl = client.delete_parameter(
+                Name=module.params.get('name')
+        )
+        changed = True
+      except botocore.exceptions.ClientError as e:
+          module.fail_json(msg=str(e))
+    return changed, nacl
+
+def main():
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name =        dict(required=True),
+        description = dict(required=False),
+        value =       dict(required=False),
+        state =       dict(default='present', choices=['present', 'absent', 'show']),
```

Comment:
Well I would at least enable check mode in the module so people can 'show' in check mode.